src/Api/IngestionBase.php

```diff
@@ -103,16 +103,16 @@ public function getUri(): string
     /**
      * @inheritDoc
      */
-    public function setMetadata(?array $metadata): IngestionInterface
+    public function setMetadata(Metadata $metadata): IngestionInterface
     {
-        $this->metadata = $metadata;
+        $this->metadata[$metadata->getKey()] = $metadata->getValue();
```

Comment:
if `$this->metadata` is `Metadata[]`, then the way you set it here is:
```suggestion
        $this->metadata = $metadata;
```